internal/middleware/authn.go

```diff
@@ -12,15 +12,21 @@ import (
 
 	"github.com/canonical/jimm/v3/internal/auth"
 	"github.com/canonical/jimm/v3/internal/errors"
-	"github.com/canonical/jimm/v3/internal/jujuapi"
 	"github.com/canonical/jimm/v3/internal/openfga"
 )
 
 // identityContextKey is the unique key to extract user from context for basic-auth authentication
 type identityContextKey struct{}
 
+// JIMMAuthner is an interface that requires authentication methods from JIMM.
+type JIMMAuthner interface {
```

Comment:
Let's avoid stripped names in types/comments.

Reply:
the idea for this proxy function is to:
- no err -> return nil
- err -> log it and try with the next one
- if you reach the end it means no controller responded you so "GATEWAY TIMEOUT"